### PR TITLE
add a link to Release Notes Language Style Guide in PR template

### LINKED ISSUE #12333: Add a link in the Release note part of the PR template
Add a link in the Release note part of the PR template

### PR DESCRIPTION
**Pull request by TomShawn**
Signed-off-by: TomShawn <41534398+TomShawn@users.noreply.github.com>

### What is changed and how it works?

Issue Number: Close #12333 

What's Changed:

```commit-message
add a link to Release Notes Language Style Guide in PR template
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch
- Related PR: https://github.com/pingcap/tidb/pull/32745

### Check List 

Tests 

- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ Side effects
 
 ### Release note <!-- bugfixes or new feature need a release note -->
 
+Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.
```
Comment (BusyJay) [inline .github/pull_request_template.md:60]: Should be included inside release-note block.

**Review by BusyJay — APPROVED** [review]

**Review by disksing — APPROVED** [review]

**Pull request merged**